pytensor/xtensor/rewriting/shape.py

```diff
@@ -164,3 +165,19 @@ def lower_expand_dims(fgraph, node):
     # Convert result back to xtensor
     result = xtensor_from_tensor(result_tensor, dims=out.type.dims)
     return [result]
+
+
+@register_lower_xtensor
+@node_rewriter(tracks=[XBroadcast])
+def lower_broadcast(fgraph, node):
+    """Rewrite XBroadcast to tensor operations."""
+    inputs = node.inputs
+
+    result_tensors = []
+
+    for x in inputs:
+        x_tensor = tensor_from_xtensor(x)
+        result = xtensor_from_tensor(x_tensor, dims=x.type.dims)
+        result_tensors.append(result)
```

Comment:
This doesn't implement broadcasting right? If there are no excluded dims, you can use `pt.broadcast_arrays` after transposing/expanding dims to align the dimensions with the output.

If there are excluded dims you have to use `pt.broadcast_to(final_shape)` for each input, comptuing the final_shape from `pt.broadcast_shape` or something like that (don't remember the exact name), after excluding the "excluded" dims from the shapes of each input.

Reply:
Again, "This is just a starting place, mostly to sketch the APIs. Don't even look at the implementations yet."

Reply:
Ah missed that. anyway, hope the pointers were useful